examples/api/lib/material/navigation_rail/navigation_rail.0.dart

```diff
@@ -92,88 +92,74 @@ class _NavRailExampleState extends State<NavRailExample> {
-                      onPressed: () {
-                        setState(() {
-                          groupAlignment = -1.0;
-                        });
-                      },
-                      child: const Text('Top'),
-                    ),
-                    ElevatedButton(
-                      onPressed: () {
-                        setState(() {
-                          groupAlignment = 0.0;
-                        });
-                      },
-                      child: const Text('Center'),
-                    ),
-                    ElevatedButton(
-                      onPressed: () {
-                        setState(() {
-                          groupAlignment = 1.0;
-                        });
-                      },
-                      child: const Text('Bottom'),
-                    ),
-                  ],
-                ),
+                SegmentedButton<double>(
+                    segments: const <ButtonSegment<double>>[
+                      ButtonSegment<double>(
+                        value: -1.0,
+                        label: Text('Top'),
+                      ),
+                      ButtonSegment<double>(
+                        value: 0.0,
+                        label: Text('Center'),
+                      ),
+                      ButtonSegment<double>(
+                        value: 1.0,
+                        label: Text('Bottom'),
+                      ),
+                    ],
+                    selected: <double>{
+                      groupAlignment
+                    },
+                    onSelectionChanged: (Set<double> newSelection) {
+                      setState(() {
+                        groupAlignment = newSelection.first;
+                      });
+                    }),
                 const SizedBox(height: 20),
-                OverflowBar(
-                  spacing: 10.0,
-                  children: <Widget>[
-                    ElevatedButton(
-                      onPressed: () {
-                        setState(() {
-                          showLeading = !showLeading;
-                        });
-                      },
-                      child: Text(showLeading ? 'Hide Leading' : 'Show Leading'),
-                    ),
-                    ElevatedButton(
-                      onPressed: () {
-                        setState(() {
-                          showTrailing = !showTrailing;
-                        });
-                      },
-                      child: Text(showTrailing ? 'Hide Trailing' : 'Show Trailing'),
-                    ),
-                  ],
+                SwitchListTile(
+                  title: Text(showLeading ? 'Hide Leading' : 'Show Leading'),
+                  value: showLeading,
+                  onChanged: (bool _) {
+                    setState(() {
+                      showLeading = !showLeading;
+                    });
+                  },
```

Comment:
```suggestion
                  onChanged: (bool value) {
                    setState(() {
                      showLeading = value;
                    });
                  },
```

I think we can assign the switch value to flip bool value.